Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs

```diff
@@ -0,0 +1,272 @@
+            if (avatarTransform.PropertiesVisible)
+            {
+                EditorGUILayout.EndFoldoutHeaderGroup();
+                EditorGUILayout.PropertyField(m_Rigidbody);
+                EditorGUILayout.PropertyField(m_PlayerInput);
+                EditorGUILayout.PropertyField(m_AvatarInputs);
+                EditorGUILayout.PropertyField(m_WalkSpeed);
+                EditorGUILayout.PropertyField(m_SprintSpeed);
+                EditorGUILayout.PropertyField(m_Acceleration);
+                EditorGUILayout.PropertyField(m_DragCoefficient);
+                EditorGUILayout.PropertyField(m_AirControlFactor);
+                EditorGUILayout.PropertyField(m_JumpImpulse);
+                EditorGUILayout.PropertyField(m_CustomGravityMultiplier);
+                EditorGUILayout.PropertyField(m_RotationSpeed);
+                EditorGUILayout.PropertyField(m_GroundCheckDistance);
+            }
+            else
+            {
+                EditorGUILayout.EndFoldoutHeaderGroup();
+            }
+
+            EditorGUILayout.Space();
+            base.OnInspectorGUI();
+        }
+    }
+#endif
+
+    [RequireComponent(typeof(Rigidbody))]
+    public class AvatarTransform : NetworkTransform
+    {
+#if UNITY_EDITOR
+        public bool PropertiesVisible = false;
+#endif
+
+        [SerializeField]
+        internal Rigidbody m_Rigidbody;
+
+        [SerializeField]
+        internal PlayerInput m_PlayerInput;
+
+        [SerializeField]
+        internal AvatarInputs m_AvatarInputs;
+
+        [SerializeField]
+        internal float m_WalkSpeed;
+
+        [SerializeField]
+        internal float m_SprintSpeed;
+
+        [SerializeField]
+        internal float m_Acceleration;
+
+        [SerializeField]
+        internal float m_DragCoefficient;
+
+        [SerializeField]
+        internal float m_AirControlFactor;
+
+        [SerializeField]
+        internal float m_JumpImpusle;
+
+        [SerializeField]
+        internal float m_CustomGravityMultiplier;
+
+        [SerializeField]
+        internal float m_RotationSpeed;
+
+        [SerializeField]
+        internal float m_GroundCheckDistance;
+
+        Vector3 m_Movement;
+
+        // grab jump state from input and clear after consumed
+        bool m_Jump;
+
+        // cached grounded check
+        bool m_IsGrounded;
+
+        RaycastHit[] m_RaycastHits = new RaycastHit[1];
+
+        Ray m_Ray;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+
+            gameObject.name = $"[Client-{OwnerClientId}]{name}";
+
+            if (!HasAuthority)
+            {
+                return;
+            }
+
+            m_PlayerInput.enabled = true;
+
+            m_Rigidbody.isKinematic = false;
+
+            // Freeze rotation on the x and z axes to prevent toppling
+            m_Rigidbody.freezeRotation = true;
+
+            var spawnPosition = new Vector3(0f, 1.5f, 0f);
```

Comment:
nitpick: there's a lot of unneeded space in this piece of code

Reply:
I've cleaned up one line, but as mentioned below, I do like spacing some bits of code out, when they're contextually different blocks of work.